src/reducers/intl.js

```diff
@@ -1,5 +1,26 @@
+import {addLocaleData} from 'react-intl';
 import {updateIntl as superUpdateIntl} from 'react-intl-redux';
-import {messages} from '../../locale/messages.json';
+import languages from '../languages.json';
+import messages from '../../locale/messages.json';
 
-export {intlReducer as default} from 'react-intl-redux';
-export const updateIntl = locale => superUpdateIntl(locale, messages[locale] || messages.en);
+export {IntlProvider, intlReducer as default} from 'react-intl-redux';
+
+Object.keys(languages).forEach(locale => {
+    // TODO: will need to handle locales not in the default intl - see www/custom-locales
+    const data = require(`react-intl/locale-data/${locale}`);
+    addLocaleData(data);
+});
+
+// start with English
+export const intlInitialState = {
+    intl: {
+        defaultLocale: 'en',
+        locale: 'en',
+        messages: messages.en
+    }
+};
+
+export const updateIntl = locale => superUpdateIntl({
```

Comment:
This may be a style preference, but I would vote for us keeping all of our exports on a single line at the end. This file exports 4 things including a default, but you have to look all over the file to see that. I would vote for something more like

```javascript
...
export {
    IntlProvider,
    intlInitialState,
    updateIntl,
    intlReducer as default
}
```

or similar. @rschamp?